tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp

```diff
@@ -733,351 +714,312 @@ getParamParentNameOffset(const ValueDecl *VD, SourceLoc Cursor) {
   return SM.getLocOffsetInBuffer(Loc, SM.findBufferContainingLoc(Loc));
 }
 
-/// Returns true on success, false on error (and sets `Diagnostic` accordingly).
-static bool passCursorInfoForDecl(SourceFile* SF,
-                                  const ValueDecl *VD,
-                                  ModuleDecl *MainModule,
-                                  const Type ContainerTy,
-                                  bool IsRef,
-                                  bool RetrieveRefactoring,
-                                  bool SymbolGraph,
-                                  ResolvedCursorInfo TheTok,
-                                  Optional<unsigned> OrigBufferID,
-                                  SourceLoc CursorLoc,
-                  ArrayRef<RefactoringInfo> KownRefactoringInfoFromRange,
-                                  SwiftLangSupport &Lang,
-                                  const CompilerInvocation &Invoc,
-                                  std::string &Diagnostic,
-                      ArrayRef<ImmutableTextSnapshotRef> PreviousASTSnaps,
-                  std::function<void(const RequestResult<CursorInfoData> &)> Receiver) {
-  if (AvailableAttr::isUnavailable(VD)) {
-    Diagnostic = "Unavailable in the current compilation context.";
-    return false;
-  }
-
-  SmallString<64> SS;
-  auto BaseType = findBaseTypeForReplacingArchetype(VD, ContainerTy);
+struct DeclInfo {
+  const ValueDecl *VD;
+  Type ContainerType;
+  bool IsRef;
+  bool IsDynamic;
+  ArrayRef<NominalTypeDecl *> ReceiverTypes;
+
+  /// If VD is a synthesized property wrapper backing storage (_foo) or
+  /// projected value ($foo) of a property (foo), the property instead.
+  /// Otherwise, VD.
+  const ValueDecl *Underlying = nullptr;
```

Comment:
Was there a problem with the old name "OriginalProperty"?  The previous name was more evocative of what it was the underlying decl *of*.

Reply:
I'll change it back 👍

Reply:
FWIW, I didn't like `OriginalProperty` because it isn't actually always the original property. It is in the $foo/_foo case, but otherwise it is just VD - which isn't always a property.